invokeai/frontend/web/src/services/events/types.ts

```diff
@@ -202,17 +209,31 @@ export type ClientEmitUnsubscribeQueue = {
   queue_id: string;
 };
 
+export type UploadImagesEvent = {
+  status: 'started' | 'processing' | 'done' | 'error';
+  message?: string;
+  progress?: number;
+  processed?: number;
+  total?: number;
+  errors?: string[];
+  images_uploading?: string[];
+  images_DTOs?: ImageDTO[];
+};
+
```

Comment:
Looks like there are 4 distinct payload types. Could be nice to have this be a discriminated union so we get good type hints when narrowing on `status`.

I'm not asking to use this feature in this PR, but [`fastapi-events`](https://github.com/melvinkcx/fastapi-events) supports pydantic models for its payloads. And a new feature lets you [dispatch specially formed pydantic models directly](https://github.com/melvinkcx/fastapi-events?tab=readme-ov-file#option-2---using-pydantic-model).

That functionality would make it easier to add these payloads to the OpenAPI schema and we'd get typescript types for free. Something to try in the future.